ops/ansible/roles/bfd-db-migrator/test/pre-test.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+# Stop immediately if any command returns a non-zero result.
+set -eou pipefail
+
+# Determine the directory that this script is in.
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Run everything from that directory.
+cd "$SCRIPT_DIR"
+
+# Create and activate the Python virtualenv needed by Ansible.
+if [ ! -d venv/ ]; then
+  python3 -m venv venv
+fi
+source venv/bin/activate
+
+# Install/upgrade wheel and pip to tamp down on noisy logs
+pip install wheel pip --upgrade
```

Comment:
Is this relatively quick if there are no updates? I know locally sometimes updating various packages takes a little bit.

Reply:
It's relatively quick. Initially I thought I'd suppress all of the log output on installations via `pip`, but that would make debugging a chore. I settled for upgrading the existing `pip` in place while installing `wheel`–strictly speaking, this **does work** without either the update or installation, it just has some unhelpful warnings for the outdated `pip` and non-existent `wheel`.